pkg/backup/backup.go

```diff
@@ -170,8 +169,22 @@ func NewKubernetesBackupper(
 
 // getNamespaceIncludesExcludes returns an IncludesExcludes list containing which namespaces to
 // include and exclude from the backup.
-func getNamespaceIncludesExcludes(backup *velerov1api.Backup) *collections.IncludesExcludes {
-	return collections.NewIncludesExcludes().Includes(backup.Spec.IncludedNamespaces...).Excludes(backup.Spec.ExcludedNamespaces...)
+func getNamespaceIncludesExcludes(backup *velerov1api.Backup, kbClient kbclient.Client) (*collections.NamespaceIncludesExcludes, []string, error) {
+	includesExcludes := collections.NewNamespaceIncludesExcludes().Includes(backup.Spec.IncludedNamespaces...).Excludes(backup.Spec.ExcludedNamespaces...)
+	nsList := corev1api.NamespaceList{}
+	activeNamespaces := []string{}
+	nsManagedByArgoCD := []string{}
```

Comment:
I suggest we isolate the managed-by-argocd logic into this new function, that is, we move the `if len(nsManagedByArgoCD) > 0` section into the current function, and not return `nsManagedByArgoCD`.

Reason: this logic is a temporary method, we may remove it after the conflict with ArgoCD is resolved. Therefore, it is better we have the logic isolated in one function, so that it won't couple with others along with the code development.

Reply:
The reason I moved it here is that my changes required going through the namespaces already -- so moving the argoCD logic here avoided having to duplicate the namespace iteration logic. I could keep it separate, but if there is a large namespace count in the cluster, there would be a performance impact.

Reply:
Understood.